script.js

```diff
@@ -110,85 +106,23 @@ next.addEventListener("click", () => {
     opt3checked.checked = false;
 
     // Remove styling for checked
-    opt0div.classList.remove("option-active");
-    opt1div.classList.remove("option-active");
-    opt2div.classList.remove("option-active");
-    opt3div.classList.remove("option-active");
+    optionDivs.forEach(div => div.classList.remove('option-active'));
 
     // selected = "";
     enabled = false;
     next.classList.remove("next-active");
   }
 });
 
-// Selection rules
-
-opt0div.addEventListener("click", () => {
-  console.log("clicked!");
-  if (!opt0div.classList.contains("option-active")) {
-    opt0div.classList.add("option-active");
-    opt1div.classList.remove("option-active");
-    opt2div.classList.remove("option-active");
-    opt3div.classList.remove("option-active");
-    opt0checked.checked = true;
-    opt1checked.checked = false;
-    opt2checked.checked = false;
-    opt3checked.checked = false;
-    selected = Questions[id].a[0].score;
-    enabled = true;
-    next.classList.add("next-active");
-    console.log("enabled? ", enabled);
-  }
-});
-opt1div.addEventListener("click", () => {
-  console.log("clicked 2!");
-  if (!opt1div.classList.contains("option-active")) {
-    opt0div.classList.remove("option-active");
-    opt1div.classList.add("option-active");
-    opt2div.classList.remove("option-active");
-    opt3div.classList.remove("option-active");
-    opt0checked.checked = false;
-    opt1checked.checked = true;
-    opt2checked.checked = false;
-    opt3checked.checked = false;
-    selected = Questions[id].a[1].score;
-    enabled = true;
-    next.classList.add("next-active");
-    console.log("enabled? ", enabled);
-  }
-});
-opt2div.addEventListener("click", () => {
-  console.log("clicked 3!");
-  if (!opt2div.classList.contains("option-active")) {
-    opt0div.classList.remove("option-active");
-    opt1div.classList.remove("option-active");
-    opt2div.classList.add("option-active");
-    opt3div.classList.remove("option-active");
-    opt0checked.checked = false;
-    opt1checked.checked = false;
-    opt2checked.checked = true;
-    opt3checked.checked = false;
-    selected = Questions[id].a[2].score;
-    enabled = true;
-    next.classList.add("next-active");
-    console.log("enabled? ", enabled);
-  }
-});
-opt3div.addEventListener("click", () => {
-  console.log("clicked 4!");
-  if (!opt3div.classList.contains("option-active")) {
-    opt0div.classList.remove("option-active");
-    opt1div.classList.remove("option-active");
-    opt2div.classList.remove("option-active");
-    opt3div.classList.add("option-active");
-    opt0checked.checked = false;
-    opt1checked.checked = false;
-    opt2checked.checked = false;
-    opt3checked.checked = true;
-    selected = Questions[id].a[3].score;
+document.querySelector('.option-container').addEventListener('click', (e) => {
+  if (e.target.nodeName === 'INPUT') {
```

Comment:
For some reason, when user clicks a label, this listener fires twice. One time for the label element and the second time for the input. We need the value from the input, so we wait for the event to be fired on the input.